Index: pool.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pool.go	(revision 6c387099ed961f654fadd34c56494a9df5ece081)
+++ pool.go	(date 1513347679000)
@@ -52,11 +52,8 @@
 
 // Shutdown releases all the pool resources.
 func (p *Pool) Shutdown() error {
-	p.rw.Lock()
-	defer p.rw.Unlock()
-
 	for _, r := range p.res {
-		r.cancel()
+		r.Release()
 	}
 
 	return nil
@@ -143,6 +140,10 @@
 
 // Release releases the pool resource.
 func (r *Res) Release() error {
+	if r.c != nil {
+		r.c.Shutdown(r.ctxt)
+	}
+
 	r.cancel()
 
 	var err error
